File: core/playlist_options.py
"""Playlist option models mirroring the Apple Music automation flags."""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import List, Optional

from config.settings import Settings


@dataclass
class PlaylistOptions:
    """Container for playlist creation preferences."""

    playlist_name: str = "Fav Artists Top Tracks"
    date_stamp: bool = False
    limit_per_artist: int = 5
    max_artists: int = 100
    max_tracks: int = 500
    dedupe_variants: bool = False
    print_tracks: bool = False
    shuffle: bool = False
    shuffle_seed: Optional[int] = None
    reuse_existing: bool = False
    truncate: bool = False
    dry_run: bool = True
    verbose: bool = False
    library_artists: bool = False
    followed_artists: bool = True
    artists_file: Optional[str] = None
    manual_artist_queries: List[str] = field(default_factory=list)
    target_playlist_id: Optional[str] = None

    @classmethod
    def from_settings(cls, settings: Settings) -> "PlaylistOptions":
        """Seed defaults from configuration settings."""
        return cls(
            playlist_name=settings.default_playlist_name,
            limit_per_artist=settings.default_limit_per_artist,
            max_artists=settings.default_max_artists,
            max_tracks=settings.default_max_tracks,
            verbose=settings.default_verbose,
        )

File: ios/Sources/DomainKit/Models/PlaylistOptions.swift
import Foundation

public struct PlaylistOptions: Codable, Equatable, Sendable {
    public var playlistName: String
    public var dateStamp: Bool
    public var limitPerArtist: Int
    public var maxArtists: Int
    public var maxTracks: Int
    public var shuffle: Bool
    public var shuffleSeed: Int?
    public var dedupeVariants: Bool
    public var reuseExisting: Bool
    public var truncate: Bool
    public var verbose: Bool
    public var dryRun: Bool
    public var preferOriginalTracks: Bool
    public var manualArtistQueries: [String]
    public var artistsFileBookmark: Data?
    public var includeLibraryArtists: Bool
    public var includeFollowedArtists: Bool

    public init(
        playlistName: String = "Untitled Playlist",
        dateStamp: Bool = true,
        limitPerArtist: Int = 5,
        maxArtists: Int = 25,
        maxTracks: Int = 100,
        shuffle: Bool = false,
        shuffleSeed: Int? = nil,
        dedupeVariants: Bool = true,
        reuseExisting: Bool = true,
        truncate: Bool = false,
        verbose: Bool = false,
        dryRun: Bool = false,
        preferOriginalTracks: Bool = true,
        manualArtistQueries: [String] = [],
        artistsFileBookmark: Data? = nil,
        includeLibraryArtists: Bool = true,
        includeFollowedArtists: Bool = false
    ) {
        self.playlistName = playlistName
        self.dateStamp = dateStamp
        self.limitPerArtist = limitPerArtist
        self.maxArtists = maxArtists
        self.maxTracks = maxTracks
        self.shuffle = shuffle
        self.shuffleSeed = shuffleSeed
        self.dedupeVariants = dedupeVariants
        self.reuseExisting = reuseExisting
        self.truncate = truncate
        self.verbose = verbose
        self.dryRun = dryRun
        self.preferOriginalTracks = preferOriginalTracks
        self.manualArtistQueries = manualArtistQueries
        self.artistsFileBookmark = artistsFileBookmark
        self.includeLibraryArtists = includeLibraryArtists
        self.includeFollowedArtists = includeFollowedArtists
    }
}
